### ISSUE TITLE
no hosts matched with inventory direcotry with script and inventory files

### ISSUE THREAD

**Issue by lufik** [ansible/ansible#4394]
I have a problem with mixing script and flat inventory files.
I created inventory script which returning empty group list and empty variables for each host.

When some conditions are met (I can't specify them - just upload small example) some hosts are excluded from all group.

Info:
$ ansible --version
ansible 1.3.2
$ rpm -qa ansible
ansible-1.3.2-1.el6.noarch
(from epel-testing)

Inventory directory: all files prefixed with 'inventory_'
Playbook file: test.yml

How to reproduce:
$ ansible-playbook test.yml --list-hosts --limit db -i ../test
$ ansible-playbook test.yml --list-hosts --limit ansible -i ../test

What output I get for both:
ERROR: provided hosts list is empty

It's ok when I remove the ad_auth or edit 'location' or 'internal'
inventory files.
When I add socond host into db-pgsql group only the second host is shown in --list-hosts.

Here is the example how can I reproduce it:
https://gist.github.com/anonymous/6874227

**Comment by jctanner** [comment]
@lufik I am not able to reproduce this on 1.3.2 or devel ...

<pre>
jtanner@u1304:~/issues/4394-inventory-file$ ansible-playbook site.yml --list-hosts --limit db -i inventory
playbook: site.yml
  play #1 (all): host count=1
    db01.domain.local
</pre>

Could you try removing the "--limit" flag and testing if the inventory is populated the way you expect?

**Comment by lufik** [comment]
Hi,
  I'm sorry for the late response but I'm on business trip so few time for even sleeping ;o)
It's working as expected without the --limit flag.

Please try to list the ansible group. It's my fault I mentioned also the db in README:
$ ansible-playbook test.yml --list-hosts --limit db -i /etc/ansible/test

playbook: test.yml

  play #1 (all): host count=1
    db01.domain.local

$ ansible-playbook test.yml --list-hosts --limit ansible -i /etc/ansible/test
ERROR: provided hosts list is empty

**Comment by jctanner** [comment]
@lufik I can reproduce the issue with "ansible" on 1.3.2 ...

```
jtanner@u1304:~/issues/4394-inventory-file$ ansible --version
ansible 1.3.2
jtanner@u1304:~/issues/4394-inventory-file$ ansible-playbook -i inventory --list-hosts --limit ansible site.yml 
ERROR: provided hosts list is empty
```

However, it is fixed as of 1.3.4 ...

```
jtanner@u1304:~/issues/4394-inventory-file$ ansible --version
ansible 1.3.4
jtanner@u1304:~/issues/4394-inventory-file$ ansible-playbook -i inventory --list-hosts --limit ansible site.yml 

playbook: site.yml

  play #1 (all): host count=1
    master01.abcd.domain.local
```

Try updating to 1.3.4 or checking out the latest devel and running your testcase with that.

**Comment by lufik** [comment]
It was fixed even in 1.3.3-1 (epel) for the example. But the root cause is still somewhere there :(
It's enough to add efgh to the master name (in internal and location files) and I have the really same problem:

master[01:01].abcd.efgh.domain.local:
$ ansible-playbook test.yml --list-hosts --limit ansible -i /etc/ansible/test
ERROR: provided hosts list is empty

And it's even more weird with
master[01:01].abcd.xyz.hgfg.domain.local:
$ ansible-playbook test.yml --list-hosts --limit ansible -i /etc/ansible/test; ansible-playbook test.yml --list-hosts --limit ansible -i /etc/ansible/test

playbook: test.yml

  play #1 (all): host count=1
    master01.abcd.xyz.hgfg.domain.local

ERROR: provided hosts list is empty

No change between two runs (and it's the same if I just fire the command twice manually). Sometimes fails the first try. Sometimes fails the second try and sometimes it works for both tries.

Unfortunately I can't check it with newer version till mid-November.
